src/dev/cpp/utils.cpp

```diff
@@ -249,7 +249,7 @@ namespace  dsn
             // optimization: zero-copy
             if (!blob.buffer_ptr())
             {
-                std::shared_ptr<char> buffer(new char[len]);
+                std::shared_ptr<char> buffer(new char[len], [](char* ptr){ delete []ptr; });
```

Comment:
There won't be a runtime cost at all as long as the compiler is sane. But using std::default_delete<T[]> might be neater.